specification/keyvault/resource-manager/Microsoft.KeyVault/preview/2021-04-01-preview/managedHsm.json

```diff
@@ -349,6 +349,157 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.KeyVault/deletedManagedHSMs": {
+      "get": {
+        "tags": [
+          "ManagedHsms"
+        ],
+        "operationId": "ManagedHsms_ListDeleted",
+        "description": "The List operation gets information about the deleted managed HSMs associated with the subscription.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Retrieved information about all managed HSMs in the specified subscription.",
+            "schema": {
+              "$ref": "#/definitions/ManagedHsmListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ManagedHsmError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List deleted managed HSMs in the specified subscription": {
+            "$ref": "./examples/DeletedManagedHsm_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.KeyVault/locations/{location}/deletedManagedHSMs/{name}": {
+      "get": {
+        "tags": [
+          "ManagedHsms"
+        ],
+        "operationId": "ManagedHsms_GetDeleted",
+        "description": "Gets the specified deleted managed HSM.",
+        "parameters": [
+          {
+            "name": "name",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the deleted managed HSM."
+          },
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The location of the deleted managed HSM."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Retrieved information about the specified deleted managed HSM.",
+            "schema": {
+              "$ref": "#/definitions/ManagedHsm"
```

Comment:
Why is this following a different model than soft delete on vaults themselves?  For vaults the deleted vaults only return some basic info about their deletion date, purge date, etc...  This returns the full resource info with one of those properties.  It should be consistent

Reply:
KeyVaults use an entirely different data object for their deleted vaults as their more complex soft deletion model required while managed hsm uses the same managed hsm object to keep track of soft deleted managed hsms. I don't believe it's appropriate here to create an entirely new data object just for soft deleted managed hsms to keep it consistent with KV since that would create unneeded complexity.

Reply:
Key vault had good reasons to do what they did in my opinion.  Doing it this way gives someone full /read access on the managedhsm as soon as its deleted if they have read on /deletedHsms.  The resource ID in /deletedHsms is also not appropriate as it does not match what was requested.  Please follow the existing vaults model as closely as possible.

Reply:
That makes sense, these changes have now been made to make managed hsm consistent with key vault.